File: Lab Week 1 Part 1/java/Polymorphism.java
import java.util.*;

public class Polymorphism {
    interface Animal {
        void speak();
    }

    static class Dog implements Animal {
        @Override public void speak() { System.out.println("Rawrr~"); }
    }

    static class Cat implements Animal {
        @Override public void speak() { System.out.println("Nyeahh~"); }
    }

    static void makeAllSpeak(List<Animal> animals) {
        for (Animal a : animals) {
            a.speak();
        }
    }

    public static void run() {
        List<Animal> animals = Arrays.asList(new Dog(), new Cat());
        makeAllSpeak(animals);
    }
}

File: Lab Week 1 Part 1/dart/bin/polymorphism.dart
abstract class Animal {
  void speak();
}

class Dog implements Animal {
  @override
  void speak() => print('Rawrr~');
}

class Cat implements Animal {
  @override
  void speak() => print('Nyeahh~');
}

void makeAllSpeak(List<Animal> animals) {
  for (var a in animals) {
    a.speak();
  }
}

void run() {
  var animals = <Animal>[Dog(), Cat()];
  makeAllSpeak(animals);
}